snapcraft/storeapi/errors.py

```diff
@@ -379,6 +379,45 @@ def __fmt_error_unknown(self, response):
         return fmt
 
 
+class StoreMetadataError(StoreError):
+
+    __FMT_NOT_REGISTERED = (
+        'Sorry, try `snapcraft register {snap_name}` before trying to '
+        'update metadata.')
```

Comment:
I'm getting this error when I try to push --only-metadata without having pushed the snap first.
We should catch when the snapcraft is registered but not pushed, and recommend to call without `--only-metadata`. Is that possible?

Reply:
Following @matthewpaulthomas's suggestion, the error messages should say what happened, why, and how to fix it.
What do you think of something like: "Failed to update metadata: snap not registered. Run first `snapcraft register {snap_name} and then `snapcraft push {snap_file}". ?

Reply:
Will change, taking also Sergio's recommendation below